libs/dal/src/repositories/message/message.repository.ts

```diff
@@ -1,25 +1,60 @@
 import { ChannelTypeEnum } from '@novu/shared';
 import { SoftDeleteModel } from 'mongoose-delete';
-import { Document, FilterQuery, Types } from 'mongoose';
+import { Document, FilterQuery, ProjectionType, Types } from 'mongoose';
 import { BaseRepository, Omit } from '../base-repository';
 import { MessageEntity } from './message.entity';
 import { Message } from './message.schema';
 import { FeedRepository } from '../feed';
-import { DalException } from '../../shared';
+import { DalException, ICacheService, Cached, InvalidateCache } from '../../shared';
 
-class PartialIntegrationEntity extends Omit(MessageEntity, ['_environmentId', '_organizationId']) {}
+class PartialMessageEntity extends Omit(MessageEntity, ['_environmentId', '_organizationId']) {}
 
-type EnforceEnvironmentQuery = FilterQuery<PartialIntegrationEntity & Document> &
+type EnforceIdentifierQuery = FilterQuery<PartialMessageEntity & Document> &
   ({ _environmentId: string } | { _organizationId: string });
 
-export class MessageRepository extends BaseRepository<EnforceEnvironmentQuery, MessageEntity> {
+type EnforceEnvironmentQuery = FilterQuery<PartialMessageEntity & Document> & { _environmentId: string } & (
```

Comment:
Not sure if this was the place where I made the suggestion but this type looks more like the `EnforceIdentifierQuery` as you are requesting the `_id` and the one that has that naming more like `EnforceEnvironmentQuery`.